MCprep_addon/spawner/mcmodel.py

```diff
@@ -175,6 +175,102 @@ def add_get_material(
 
 	return mat
 
+def find_all_pack_roots(
+	start_path):
+	"""
+	Finds ALL possible pack roots, including:
+		- normal packs with 'assets/'
+		- loose/unpacked mod folders containing `models/` and `textures/`
+		- nested packs
+
+	A pack root is ANY folder that contains:
+		• assets/<namespace>/...
+		OR
+		• textures/	  (for packs missing the assets folder)
+		• models/	  (same case)
+	"""
+	roots = []
+	start = Path(start_path).resolve()
+
+	for parent in [start] + list(start.parents):
+
+		# Case A: vanilla-style root
+		if (parent / "assets").is_dir():
+			roots.append(parent)
+
+		# Case B: loose/unpacked mod folder (your case)
+		# e.g. christmasfestivity/textures/block/*.png
+		if (parent / "textures").is_dir() and (parent / "models").is_dir():
+			roots.append(parent)
+
+		# Case C: even looser: any folder with a textures/ directory
+		if (parent / "textures").is_dir():
+			roots.append(parent)
+
+	# Make unique and preserve order
+	seen = set()
+	final = []
+	for r in roots:
+		if r not in seen:
+			seen.add(r)
+			final.append(r)
```

Comment:
More of a style comment, but the condition should be reversed to avoid excessive indentation.